pkg/server/api/bundle/v1/service.go

```diff
@@ -0,0 +1,158 @@
+}
+
+func (s *Service) GetFederatedBundle(ctx context.Context, req *bundle.GetFederatedBundleRequest) (*types.Bundle, error) {
+	log := rpccontext.Logger(ctx).WithField("RPC", "GetFedertedBundle")
+
+	if !(rpccontext.CallerIsLocal(ctx) || rpccontext.CallerIsAdmin(ctx) || rpccontext.CallerIsAgent(ctx)) {
+		log.Errorf("Permission denied: the caller must be local or present an admin or an active agent X509-SVID")
+		return nil, status.Errorf(codes.PermissionDenied, "the caller must be local or present an admin or an active agent X509-SVID")
+	}
+
+	td, err := spiffeid.TrustDomainFromString(req.TrustDomain)
+	if err != nil {
+		log.Errorf("Trust domain argument is not a valid SPIFFE ID: %q", req.TrustDomain)
+		return nil, status.Errorf(codes.InvalidArgument, "trust domain argument is not a valid SPIFFE ID: %q", req.TrustDomain)
+	}
+
+	if s.td.Compare(td) == 0 {
+		log.Errorf("%q is this server own trust domain, use GetBundle RPC instead", td.String())
+		return nil, status.Errorf(codes.InvalidArgument, "%q is this server own trust domain, use GetBundle RPC instead", td.String())
+	}
+
+	dsResp, err := s.ds.FetchBundle(ctx, &datastore.FetchBundleRequest{
+		TrustDomainId: td.IDString(),
+	})
+	if err != nil {
+		log.Errorf("Failed to fetch bundle: %v", err)
+		return nil, status.Errorf(codes.Internal, "failed to fetch bundle: %v", err)
+	}
+
+	if dsResp.Bundle == nil {
+		log.Errorf("Bundle for %q not found", req.TrustDomain)
+		return nil, status.Errorf(codes.NotFound, "bundle for %q not found", req.TrustDomain)
+	}
+
+	b, err := applyMask(dsResp.Bundle, req.OutputMask)
+	if err != nil {
+		log.Errorf("Failed to apply mask: %v", err)
+		return nil, status.Errorf(codes.Internal, "failed to apply mask: %v", err)
+	}
+
+	return b, nil
+}
+
+func (s *Service) BatchCreateFederatedBundle(ctx context.Context, req *bundle.BatchCreateFederatedBundleRequest) (*bundle.BatchCreateFederatedBundleResponse, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method BatchCreateFederatedBundle not implemented")
+}
+
+func (s *Service) BatchUpdateFederatedBundle(ctx context.Context, req *bundle.BatchUpdateFederatedBundleRequest) (*bundle.BatchUpdateFederatedBundleResponse, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method BatchUpdateFederatedBundle not implemented")
+}
+
+func (s *Service) BatchSetFederatedBundle(ctx context.Context, req *bundle.BatchSetFederatedBundleRequest) (*bundle.BatchSetFederatedBundleResponse, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method BatchSetFederatedBundle not implemented")
+}
+
+func (s *Service) BatchDeleteFederatedBundle(ctx context.Context, req *bundle.BatchDeleteFederatedBundleRequest) (*bundle.BatchDeleteFederatedBundleResponse, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method BatchDeleteFederatedBundle not implemented")
+}
+
+func applyMask(b *common.Bundle, mask *types.BundleMask) (*types.Bundle, error) {
```

Comment:
I missed this in my first pass. Output mask is optional. If unset, it means "return everything".